src/main/kotlin/io/sc3/peripherals/posters/PosterPrintData.kt

```diff
@@ -0,0 +1,94 @@
+package io.sc3.peripherals.posters
+
+import io.sc3.library.ext.optString
+import io.sc3.library.ext.putOptString
+import io.sc3.peripherals.config.ScPeripheralsConfig.config
+import io.sc3.peripherals.mixin.MapColorAccessor
+import io.sc3.peripherals.posters.PosterItem.Companion.POSTER_KEY
+import net.minecraft.block.MapColor
+import net.minecraft.nbt.NbtCompound
+import net.minecraft.text.Text
+import kotlin.math.roundToInt
+import kotlin.math.sign
+
+const val MAX_LABEL_LENGTH = 48
+const val MAX_TOOLTIP_LENGTH = 256
+
+fun getDefaultPalette() = MapColorAccessor.getColors().map { it?.color ?: MapColor.CLEAR.color }.toIntArray()
+
+data class PosterPrintData(
+  private val initialLabel: String? = null,
+  var tooltip: String? = null,
+  val colors: ByteArray = ByteArray(128 * 128) { 0 },
```

Comment:
Same here re: all zeroes (not sure if this comment will appear before the other one)

```suggestion
  val colors: ByteArray = ByteArray(128 * 128),
```